VestingFactory.sol

```diff
@@ -0,0 +1,55 @@
+pragma solidity ^0.5.3;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+import "../common/Initializable.sol";
+import "../common/UsingRegistry.sol";
+import "./VestingSchedule.sol";
+
+// Revision 1
+contract VestingFactory is ReentrancyGuard, Initializable, UsingRegistry {
+
+    // mapping between beneficiary addresses and associated vesting contracts (schedules)
+    mapping(address => VestingSchedule) private vestingSchedules;
+
+    function initialize(address registryAddress) external initializer {
+      _transferOwnership(msg.sender);
+      setRegistry(registryAddress);
```

Comment:
What is the idea behind `setRegistry`? Is it something celo specific

Reply:
since its a protocol contract, that is required for it to be registred